iidm/iidm-impl/src/main/java/com/powsybl/iidm/network/impl/ShuntCompensatorImpl.java

```diff
@@ -107,19 +130,88 @@ public double getMaximumB() {
         return bPerSection * maximumSectionCount;
     }
 
+    @Override
+    public TerminalExt getRegulatingTerminal() {
+        return regulatingTerminal;
+    }
+
+    @Override
+    public ShuntCompensatorImpl setRegulatingTerminal(Terminal regulatingTerminal) {
+        ValidationUtil.checkRegulatingTerminal(this, (TerminalExt) regulatingTerminal, getNetwork());
```

Comment:
Are you sure about this control: should we get a valid terminal:
```
this.regulatingTerminal = regulatingTerminal != null ? (TerminalExt) regulatingTerminal : getTerminal();
```

Reply:
Not sure, the check is taking place if the terminal is not null anyway.